numba/core/typed_passes.py

```diff
@@ -289,6 +289,14 @@ def run_pass(self, state):
         """
         Convert data-parallel computations into Parfor nodes
         """
+        # Register lowerer for Parfor Node
+        from numba.parfors.parfor_lowering import _lower_parfor_parallel
+        if hasattr(state.targetctx, "lower_extensions"):
+            state.targetctx.lower_extensions[Parfor] = _lower_parfor_parallel
```

Comment:
Can we avoid assigning to the target context in the middle the pipeline?

Reply:
I absolutely wanted to, but could not find a better place. Please suggest. My thought process: the pass that adds the `special_extension_ir_node` should also add how to lower it.

Reply:
I think it make sense for the ``CPUContext.init`` to also insert the ``Parfor`` lowering.

Reply:
Should we be inserting the `Parfor` lowering in both places? Or should the initialization be removed from this pass and only be happening in `CPUContext.init`?

Reply:
Yes, remove from this pass and only in `CPUContext.init`.

Reply:
Done.